### PR TITLE
Fix create source distribution step for release

### PR DESCRIPTION
**Pull request by atalman**
This is fixing following failure in the release branch:
```
cp: cannot create directory '/tmp/pytorch-release/2.1': No such file or directory
```
Link: https://github.com/pytorch/pytorch/actions/runs/6591657669/job/17910724990

cp will report that error if the parent directory (pytorch-release in this case) does not exist. 
This is working in main since ``PT_RELEASE_NAME: pytorch-main`` however for release its ``PT_RELEASE_NAME: pytorch-release/2.1`` 

Test:
```
export tag_or_branch=release/2.1
tag_or_branch="${tag_or_branch//\//_}"
echo $tag_or_branch 
release_2.1
```

### CODE REVIEW

**Review by huydhn — APPROVED** [review]

**Review by osalpekar — APPROVED** [review]

File: .github/workflows/create_release.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ jobs:
         run: |
             # Create new folder with specified name so extracting the archive yields that
             rm -rf "/tmp/$PT_RELEASE_NAME"
+            mkdir "/tmp/$PT_RELEASE_NAME"
```
Comment (malfet) [inline .github/workflows/create_release.yml:39]: Shouldn't it be `-p`?
```suggestion
            mkdir -p "/tmp/$PT_RELEASE_NAME"
```

**Review by malfet — CHANGES_REQUESTED** [review]
Can you please elaborate, what is the desired outcome here?

**Review by malfet — COMMENTED** [review]
What you really want to do here, is to add `tag_or_branch="${tag_or_branch//\//_}"` line to "Set filenames" step, as the goal is to remove directory separators fro the branch names...

File: .github/workflows/create_release.yml (hunk 1 above)
Comment (atalman) [reply .github/workflows/create_release.yml]: Used suggested approach: `` tag_or_branch="${tag_or_branch//\//_}"``

**Pull request closed without merging**